pkg/resource/license.go

```diff
@@ -0,0 +1,119 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package resource
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	commonlicense "github.com/elastic/cloud-on-k8s/pkg/controller/common/license"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/version"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/license"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// defaultOperatorLevel is the default license level when no operator license is installed
+	defaultOperatorLevel = "basic"
+	// minSupportedVersion is the minimum Elastic stack version used to find an operator license
+	minSupportedVersion = "6.8.0"
+	// licensingCfgMapName is the name of the config map used to store licensing information
+	licensingCfgMapName = "elastic-licensing"
+)
+
+// LicensingInfo represents information about the operator license including the total memory of all Elastic managed
+// components
+type LicensingInfo struct {
+	Timestamp               string `json:"timestamp"`
+	LicenseLevel            string `json:"license_level"`
+	MemoryInGiga            string `json:"memory_in_giga"`
```

Comment:
I did this thinking that it could ease potential integrations by external systems (inspired by prometheus, e.g.: `go_memstats_sys_bytes: 7.3335032e+07`), but
```json
"memory": "5.37GB"
```
also looks fine to me.